From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 21 Jan 2026 22:00:17 +0100
Subject: guniprop: Use size_t for output_marks length

The input string length may overflow, and this would lead to wrong
behavior and invalid writes.

Spotted by treeplus.
Thanks to the Sovereign Tech Resilience programme from the Sovereign
Tech Agency.

ID: #YWH-PGM9867-171
Bug: https://gitlab.gnome.org/GNOME/glib/-/issues/3872
Bug-CVE: CVE-2026-1489
Bug-Debian: https://bugs.debian.org/1126549
Applied-upstream: 2.87.3, commit:1d4052c5a5e4504a6906fda12ad4a3f4fb1c9233
---
 glib/guniprop.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/glib/guniprop.c b/glib/guniprop.c
index a3af532..0ee9aa3 100644
--- a/glib/guniprop.c
+++ b/glib/guniprop.c
@@ -772,13 +772,13 @@ get_locale_type (void)
   return LOCALE_NORMAL;
 }
 
-static gint
+static size_t
 output_marks (const char **p_inout,
 	      char        *out_buffer,
 	      gboolean     remove_dot)
 {
   const char *p = *p_inout;
-  gint len = 0;
+  size_t len = 0;
   
   while (*p)
     {
